src/main/java/org/sagebionetworks/web/client/presenter/EmailInvitationPresenter.java

```diff
@@ -0,0 +1,158 @@
+import org.sagebionetworks.web.shared.NotificationTokenType;
+
+import javax.inject.Inject;
+
+import static org.sagebionetworks.web.client.DisplayUtils.getDisplayName;
+
+public class EmailInvitationPresenter extends AbstractActivity implements EmailInvitationView.Presenter, Presenter<EmailInvitation> {
+	private EmailInvitationView view;
+	private RegisterWidget registerWidget;
+	private SynapseClientAsync synapseClient;
+	private SynapseAlert synapseAlert;
+	private AuthenticationController authController;
+	private PlaceChanger placeChanger;
+
+	@Inject
+	public EmailInvitationPresenter(EmailInvitationView view,
+									RegisterWidget registerWidget,
+									SynapseClientAsync synapseClient,
+									SynapseAlert synapseAlert,
+									AuthenticationController authController,
+									GlobalApplicationState globalApplicationState) {
+		this.view = view;
+		this.registerWidget = registerWidget;
+		this.synapseClient = synapseClient;
+		this.synapseAlert = synapseAlert;
+		this.authController = authController;
+		this.placeChanger = globalApplicationState.getPlaceChanger();
+		view.setPresenter(this);
+		registerWidget.enableEmailAddressField(false);
+		view.setRegisterWidget(registerWidget.asWidget());
+		view.setSynapseAlertContainer(synapseAlert.asWidget());
+	}
+
+	@Override
+	public void start(AcceptsOneWidget acceptsOneWidget, EventBus eventBus) {
+		acceptsOneWidget.setWidget(view);
+	}
+
+	@Override
+	public void setPlace(EmailInvitation place) {
+	    final String encodedMembershipInvtnSignedToken = place.toToken();
+		PromiseRPC<SignedTokenInterface> promise = new PromiseRPC<>();
+		synapseClient.hexDecodeAndDeserialize(NotificationTokenType.EmailInvitation.name(), encodedMembershipInvtnSignedToken, promise);
+		promise.then(new Function() {
+			@Override
+			public PromiseRPC<MembershipInvtnSubmission> f(Object... args) {
+				MembershipInvtnSignedToken token = (MembershipInvtnSignedToken) args[0];
+				PromiseRPC<MembershipInvtnSubmission> promise = new PromiseRPC<>();
+				synapseClient.getMembershipInvitation(token, promise);
+				return promise;
+			}
+		}).done(new Function() {
+			@Override
+			public void f() {
+				MembershipInvtnSubmission mis = (MembershipInvtnSubmission) arguments(0);
+				if (authController.isLoggedIn()) {
+					bindInvitationToAuthenticatedUser(mis.getId());
+				} else {
+					EmailInvitationPresenter.this.registerWidget.setEncodedMembershipInvtnSignedToken(encodedMembershipInvtnSignedToken);
+					EmailInvitationPresenter.this.registerWidget.setEmail(mis.getInviteeEmail());
+					setupInvitationMessage(mis);
+				}
+			}
+		}).fail(new Function() {
+			@Override
+			public void f() {
+				Throwable throwable = arguments(0);
+				synapseAlert.handleException(throwable);
+			}
+		});
+	}
+
+	private void bindInvitationToAuthenticatedUser(final String misId) {
+		PromiseRPC<InviteeVerificationSignedToken> promise = new PromiseRPC<>();
```

Comment:
Please format new code, indentation should be done in either tabs and spaces (I think default for Eclipse Format or Correct Indentation uses tabs).

Reply:
This should be fixed